funnel/views/commentvote.py

```diff
@@ -127,6 +131,60 @@ def parent_comments_url(obj):
     return url
 
 
+@route('/comments')
+class AllCommentsView(ClassView):
+    current_section = 'comments'
+
+    @route('', endpoint='comments')
+    @requires_login
+    @render_with(json=True)
+    def view(self):
+        commentset_memberships = [
+            {
+                'parent': cm.commentset.parent.current_access(
+                    datasets=('primary', 'related')
+                ),
+                'commentset_url': cm.commentset.url_for(_external=True),
+                'last_seen_at': cm.last_seen_at,
+                'new_comments_count': (
+                    cm.commentset.comments.filter(Comment.state.PUBLIC)
+                    .filter(Comment.created_at > cm.last_seen_at)
+                    .count()
+                ),
+                'last_comment': (
+                    cm.commentset.comments.filter(Comment.state.PUBLIC)
+                    .order_by(Comment.created_at.desc())
+                    .first()
+                    .current_access(datasets=('primary', 'related'))
+                ),
+            }
```

Comment:
These nested queries are murder on performance. They need to be:
1. Loaded at once using a single joined query, or
2. Loaded from cache, or
3. Specifically requested by the client one at a time.